console/collections/src/merkle_tree/mod.rs

```diff
@@ -721,28 +721,15 @@ impl<E: Environment, LH: LeafHash<Hash = PH::Hash>, PH: PathHash<Hash = Field<E>
 
 /// Returns the depth of the tree, given the size of the tree.
 #[inline]
-#[allow(clippy::cast_possible_truncation)]
 fn tree_depth<const DEPTH: u8>(tree_size: usize) -> Result<u8> {
     let tree_size = u64::try_from(tree_size)?;
-    // Ensure the tree size is less than 2^52 (for casting to an f64).
-    let tree_depth = match tree_size < 4503599627370496_u64 {
-        // Compute the log2 of the tree size.
-        true => (tree_size as f64).log2(),
-        false => bail!("Tree size must be less than 2^52"),
-    };
-    // Ensure the tree depth is within a u8 range.
-    match tree_depth <= u8::MAX as f64 {
-        true => {
-            // Convert the tree depth to a u8.
-            let tree_depth = tree_depth as u8;
-            // Ensure the tree depth is within the depth bound.
-            match tree_depth <= DEPTH {
-                // Return the tree depth.
-                true => Ok(tree_depth),
-                false => bail!("Merkle tree cannot exceed depth {DEPTH}: attempted to reach depth {tree_depth}"),
-            }
-        }
-        false => bail!("Merkle tree depth ({tree_depth}) exceeds maximum size ({})", u8::MAX),
+    // Since we only allow tree sizes up to u64::MAX, the maximum possible depth is 63.
+    let tree_depth = u8::try_from(tree_size.ilog2())?;
```

Comment:
`ilog2` will panic if `tree_size` is 0. https://doc.rust-lang.org/std/primitive.u64.html#method.ilog2

We may want to use `checked_ilog2`

Reply:
what should we do in case the `tree_size` is `0`? throw an error or determine that the `tree_depth` is `0`? I assume the latter, as it's closer to what we have right now (`-inf as u8` is `0`)